### DISCUSSION TITLE
Problems building on Mac: homebrew errors and can't find freetype

### DISCUSSION THREAD

**Discussion by surferjeff** [ImageMagick/ImageMagick#2314] [Help] [answered]
I'm trying to build ImageMagick from source on Mac OS 10.15.5, with the options --with-quantum-depth=8 --disable-hdri 

I followed the instructions here:
https://github.com/ImageMagick/ImageMagick/blob/master/Install-mac.txt

The homebrew instructions didn't work for me:
```
jeff@MacBook-Pro Development % brew options imagemagick
jeff@MacBook-Pro Development % brew reinstall imagemagick --with-quantum-depth=8 --disable-hdri 
Usage: brew reinstall [options] formula

Uninstall and then install formula using the same options it was originally
installed with, plus any appended brew formula options.

Unless HOMEBREW_NO_INSTALL_CLEANUP is set, brew cleanup will then be run for
the reinstalled formulae or, every 30 days, for all formulae.

    -d, --debug                      If brewing fails, open an interactive
                                     debugging session with access to IRB or a
                                     shell inside the temporary build directory.
    -s, --build-from-source          Compile formula from source even if a
                                     bottle is available.
    -i, --interactive                Download and patch formula, then open a
                                     shell. This allows the user to run
                                     ./configure --help and otherwise
                                     determine how to turn the software package
                                     into a Homebrew package.
        --force-bottle               Install from a bottle if it exists for the
                                     current or newest version of macOS, even if
                                     it would not normally be used for
                                     installation.
        --keep-tmp                   Retain the temporary files created during
                                     installation.
    -f, --force                      Install without checking for previously
                                     installed keg-only or non-migrated
                                     versions.
    -v, --verbose                    Print the verification and postinstall
                                     steps.
        --display-times              Print install times for each formula at the
                                     end of the run.
    -h, --help                       Show this message.
Error: invalid option: --with-quantum-depth=8
```

So, I cloned this repo and checked out tag 7.0.10-23.

The following commands succeeded:
```
    $magick> sudo port -v install freetype +bytecode
    $magick> sudo port -v install librsvg
    $magick> sudo port -v install graphviz +gs +wmf +jbig +jpeg2 +lcms
    $magick> sudo port install jpeg
```

But no matter how I run `./configure`, it never finds the freetype library:
```
    $magick> ./configure --prefix=/Users/jeff/gitrepos/ImageMagick/build --with-quantum-depth=8 --disable-hdri \
      --disable-dependency-tracking --without-perl


...

-------------------------------------------------------------
checking for freetype2... no

...

Delegate library configuration:
  FreeType          --with-freetype=yes		no

```

What am I doing wrong?

My ultimate goal is to be able -annotate jpeg and tiff files with the `--with-quantum-depth=8 --disable-hdri` flags set for quicker execution.

Thanks for your help!

**Comment by surferjeff** [comment, marked as answer]
Homebrew no longer supports options:
https://brew.sh/2019/02/02/homebrew-2.0.0/

I found an alternative:
```
$> homebrew edit imagemagick
$> homebrew reinstall -s imagemagick
```

**Comment by hammersxo** [comment]
> I'm trying to build ImageMagick from source on Mac OS 10.15.5, with the options --with-quantum-depth=8 --disable-hdri
> 
> I followed the instructions here:
> https://github.com/ImageMagick/ImageMagick/blob/master/Install-mac.txt
> 
> The homebrew instructions didn't work for me:
> 
> ```
> jeff@MacBook-Pro Development % brew options imagemagick
> jeff@MacBook-Pro Development % brew reinstall imagemagick --with-quantum-depth=8 --disable-hdri 
> Usage: brew reinstall [options] formula
> 
> Uninstall and then install formula using the same options it was originally
> installed with, plus any appended brew formula options.
> 
> Unless HOMEBREW_NO_INSTALL_CLEANUP is set, brew cleanup will then be run for
> the reinstalled formulae or, every 30 days, for all formulae.
> 
>     -d, --debug                      If brewing fails, open an interactive
>                                      debugging session with access to IRB or a
>                                      shell inside the temporary build directory.
>     -s, --build-from-source          Compile formula from source even if a
>                                      bottle is available.
>     -i, --interactive                Download and patch formula, then open a
>                                      shell. This allows the user to run
>                                      ./configure --help and otherwise
>                                      determine how to turn the software package
>                                      into a Homebrew package.
>         --force-bottle               Install from a bottle if it exists for the
>                                      current or newest version of macOS, even if
>                                      it would not normally be used for
>                                      installation.
>         --keep-tmp                   Retain the temporary files created during
>                                      installation.
>     -f, --force                      Install without checking for previously
>                                      installed keg-only or non-migrated
>                                      versions.
>     -v, --verbose                    Print the verification and postinstall
>                                      steps.
>         --display-times              Print install times for each formula at the
>                                      end of the run.
>     -h, --help                       Show this message.
> Error: invalid option: --with-quantum-depth=8
> ```
> 
> So, I cloned this repo and checked out tag 7.0.10-23.
> 
> The following commands succeeded:
> 
> ```
>     $magick> sudo port -v install freetype +bytecode
>     $magick> sudo port -v install librsvg
>     $magick> sudo port -v install graphviz +gs +wmf +jbig +jpeg2 +lcms
>     $magick> sudo port install jpeg
> ```
> 
> But no matter how I run `./configure`, it never finds the freetype library:
> 
> ```
>     $magick> ./configure --prefix=/Users/jeff/gitrepos/ImageMagick/build --with-quantum-depth=8 --disable-hdri \
>       --disable-dependency-tracking --without-perl
> 
> 
> ...
> 
> -------------------------------------------------------------
> checking for freetype2... no
> 
> ...
> 
> Delegate library configuration:
>   FreeType          --with-freetype=yes		no
> ```
> 
> What am I doing wrong?
> 
> My ultimate goal is to be able -annotate jpeg and tiff files with the `--with-quantum-depth=8 --disable-hdri` flags set for quicker execution.
> 
> Thanks for your help!

Thanks for updating the thread and posting the alternatives.
